services/galley/src/Galley/Data.hs

```diff
@@ -309,6 +310,11 @@ userTeams u =
   map runIdentity
     <$> retry x1 (query Cql.selectUserTeams (params Quorum (Identity u)))
 
+usersTeams :: MonadClient m => [UserId] -> m (Map UserId TeamId)
+usersTeams uids = do
+  pairs <- retry x1 (query Cql.selectUsersTeams (params Quorum (Identity uids)))
```

Comment:
Larger IN queries (I know there's currently chunks of 32 in use; but still it's not ideal even at 32) lead to performance problems. See https://lostechies.com/ryansvihla/2014/09/22/cassandra-query-patterns-not-using-the-in-query-for-multiple-partitions/

Preferred way to do this lookup here is to use `pooledMapConcurrentlyN` (https://hoogle.zinfra.io/file/root/.stack/snapshots/x86_64-linux/e2cc9ab01ac828ffb6fe45a45d38d7ca6e672fb9fe95528498b990da673c5071/8.8.4/doc/unliftio-0.2.13/UnliftIO-Async.html#v:pooledMapConcurrentlyN) (e.g. with 8 threads). Then you can re-use the existing `selectOneUserTeam`.

Okay to do this in a follow-up PR if you prefer (but please do this change before this all goes live).